tests/strategies/test_deepspeed_strategy.py

```diff
@@ -232,7 +232,8 @@ def backward(self, loss: Tensor, optimizer: Optimizer, optimizer_idx: int, *args
     [(RandomDataset, "auto"), (RandomDataset, 10), (RandomIterableDataset, "auto"), (RandomIterableDataset, 10)],
 )
 @mock.patch("deepspeed.init_distributed", autospec=True)
-def test_deepspeed_auto_batch_size_config_select(mock_deepspeed_distributed, tmpdir, dataset_cls, value):
+@mock.patch("pytorch_lightning.Trainer.log_dir", new_callable=mock.PropertyMock, return_value="abc")
```

Comment:
Why did you need to add this?

Reply:
It is because profiler need to access `log_dir` and broadcast `log_dir` through `DeepSpeedStrategy`. And will raise "distributed not init" error because the  `deepspeed.init_distributed` is mocked in this test. So we mock the log_dir too.